go/vt/vtgate/planbuilder/system_tables.go

```diff
@@ -25,7 +27,7 @@ import (
 )
 
 func (pb *primitiveBuilder) findSysInfoRoutingPredicates(expr sqlparser.Expr, rut *route) error {
-	tableSchemas, tableNames, err := extractInfoSchemaRoutingPredicate(expr)
+	tableSchemas, tableNames, err := extractInfoSchemaRoutingPredicate(expr, rut.eroute.SysTableTableSchema, rut.eroute.SysTableTableName)
```

Comment:
from my understanding this SysTableTableSchema will always be empty at this step.

Reply:
ok, I saw you used this in the or statement below. Please make a comment here.